backend/assignments/1/src/main/java/org/sanskar/Main.java

```diff
@@ -0,0 +1,132 @@
+package org.sanskar;
+import org.sanskar.coin.Coin;
+import org.sanskar.json.JsonReader;
+import org.sanskar.trader.Trader;
+import org.sanskar.coin.coinHelper;
+import org.sanskar.trader.tradingHelper;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.sanskar.csvReader.*;
+import java.util.logging.Logger;
+
+
+public class Main {
+
+    private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+
+    private static ExecutorService executorService;
+    private static List<Coin> coins;
+    private static List<Trader> traders;
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+
+        String coinPath = "src/main/resources/coins.csv";
+        String traderPath = "src/main/resources/traders.csv";
+        String jsonPath = "src/main/resources/test_transaction.json";
+
+        // Loading csv files
+        coins = readCoins(coinPath);
+        traders = readTraders(traderPath);
+
+//-------------------------------------------------------
+        int topN = 5;  // Adjust as needed
+
+        LOGGER.info("-------------------------------------------------");
+        // Get top N coins
+        List<Coin> topCoins = coinHelper.getTopCoins(coins, topN);
+
+        // Show portfolio for a trader
+        Trader trader = traders.get(0);
+        tradingHelper.showPortfolio(trader);
+
+        // Get top N traders
+        List<Trader> topTraders = LeaderboardService.getTopTraders(traders, topN);
+
+        // Get bottom N traders
+        List<Trader> bottomTraders = LeaderboardService.getBottomTraders(traders, topN);
+        LOGGER.info("Top N Coins:");
+        for (Coin coin : topCoins) {
+            LOGGER.info(coin.getName() + " - " + coin.getPrice());
+        }
+        LOGGER.info("-------------------------------------------------");
+        LOGGER.info("\nTrader Portfolio:");
+        tradingHelper.showPortfolio(trader);
+        LOGGER.info("-------------------------------------------------");
+        LOGGER.info("\nTop N Traders:");
+        for (Trader topTrader : topTraders) {
+            LOGGER.info(topTrader.getFirstName() + " " + topTrader.getLastName());
+        }
+        LOGGER.info("-------------------------------------------------");
+        LOGGER.info("\nBottom N Traders:");
+        for (Trader bottomTrader : bottomTraders) {
+           LOGGER.info(bottomTrader.getFirstName() + " " + bottomTrader.getLastName());
+        }
+        LOGGER.info("-------------------------------------------------");
```

Comment:
You were supposed to take user input for number of top/bottom traders, portfolio